webapp/interop_handler_test.go

```diff
@@ -0,0 +1,66 @@
+//go:build small
+// +build small
+
+package webapp
+
+// Copyright 2022 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInteropHandler_redirect(t *testing.T) {
+	// 1999 is an invalid interop year and should be redirected.
+	req := httptest.NewRequest("GET", "/interop-1999?embedded", strings.NewReader("{}"))
+	req = mux.SetURLVars(req, map[string]string{
+		"name":     "interop",
+		"year":     "1999",
+		"embedded": "true",
+	})
+
+	w := httptest.NewRecorder()
+	interopHandler(w, req)
+	resp := w.Result()
+	assert.Equal(t, resp.StatusCode, http.StatusTemporaryRedirect)
+
+	loc, err := resp.Location()
+	assert.Nil(t, err)
+	// Check if embedded param is maintained after redirect.
+	assert.NotEqual(t, loc.Path, "interop-"+defaultRedirectYear+"?embedded")
```

Comment:
For this test case assertion, you are testing that it went to the default year. you should hardcode the year here. In the case that the default year changes, this test should break and we should then update the value here.

`"interop-"+defaultRedirectYear+"?embedded"` -> `"interop-2021?embedded"`

Reply:
Smart idea, I've changed it 🙂